docs/DeveloperGuide.md

```diff
@@ -320,9 +320,26 @@ The find operation is facilitated by `FindCommand`. It extends `Command` and imp
 
 Given below is an example usage scenario and how the find command operates in TrackAScholar:
 
-1. The user enters `find john`.The application searches the data for all the applicants with the name that matches
+1. The user enters `find n/john` into the terminal.
+   This invokes `LogicManager#execute()`, which calls `TrackAScholarParser#parseCommand()` to separate the command word `find` and
+   the arguments `n/ john`.
 
-2. `TrackAScholarParser` identifies the person and displays it to the user.
+2. `TrackAScholarParser` identifies the `find` command and argument `n/john`. `FindCommandParser` will be instantiated which calls `FindCommandParser#parse()`
+    
+3. `FindCommandParser#parse()` first checks for a valid prefix which can be of type name, scholarship or major.It throws out an error if a valid prefix is not found.
+
+4. Since a valid prefix for name is present `FindCommandParser#parse()` will then call `parsePredicates` with the argument `john`.
+
+5. In `parsePredicates` an empty is first created `applicantPredicateList` before, it checks for the type of prefix accompanying the argument which be of type  name, scholarship or major.
+
+6. After identifying the prefix is of type name it will call `getKeywordsList` to get the keyword for searching and in this case it will be `john`.
+
+7. After identifying the keyword `john` the method, `NameContainsKeywordsPredicate` will be invoked with the argument `john`.
+
+8. `NameContainsKeywordsPredicate` will check if any names in the main list has `john` present in their names.If `john` is present, that name in the main
+    list will be added to `applicantPredicateList`.All valid names are added to `applicantPredicateList`
```

Comment:
Try to abstract it out more, dont speak about specific prefixes or keywords too much, check the edit applicant feature below for an example

Reply:
Tried to work on it